### PR TITLE
Throwing error on empty csv or json file

### PR DESCRIPTION
**Pull request by sukaant**

### CODE REVIEW

**Review by dunson062786 — APPROVED** [review]

File: apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex [hunk 1]
```diff
@@ -167,12 +167,18 @@ defmodule AndiWeb.EditLiveView.DataDictionaryForm do
...
-    new_changeset =
-      file
-      |> parse_csv()
-      |> DataDictionaryFormSchema.changeset_from_tuple_list(socket.assigns.dataset_id)
-
-    assign_new_schema(socket, new_changeset)
+    try do
+      new_changeset =
+        file
+        |> String.equivalent?("\n")
```
Comment (LtChae) [inline apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex:173]: Will this handle legitimately empty files? i.e. no whitespace no nothing.

File: apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex [hunk 2]
```diff
@@ -167,12 +167,18 @@ defmodule AndiWeb.EditLiveView.DataDictionaryForm do
...
+        |> if(do: throw(:error), else: file)
```
Comment (JarredOlson) [inline apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex:174]: Please validate this change with a unit or integration test.

File: apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex [hunk 3]
```diff
@@ -182,30 +188,28 @@ defmodule AndiWeb.EditLiveView.DataDictionaryForm do
...
-        new_changeset =
-          decoded_json
-          |> List.wrap()
-          |> DataDictionaryFormSchema.changeset_from_file(socket.assigns.dataset_id)
-
-        assign_new_schema(socket, new_changeset)
+        try do
+          new_changeset =
+            decoded_json
+            |> Enum.empty?()
```
Comment (JarredOlson) [inline apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex:198]: Please validate this change with a unit or integration test.

**Review by JarredOlson — CHANGES_REQUESTED** [review]

File: apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex [hunk 4]
```diff
@@ -217,12 +221,8 @@ defmodule AndiWeb.EditLiveView.DataDictionaryForm do
...
 
   def handle_event("overwrite-schema", _, %{assigns: %{pending_changeset: nil}} = socket) do
-    changeset_with_error =
-      socket.assigns.changeset
-      |> Map.put(:action, :update)
-      |> Changeset.add_error(:schema_sample, "There was a problem interpreting this file")
-
-    {:noreply, assign(socket, changeset: changeset_with_error)}
+    socket.assigns.changeset
+    |> send_error_interpreting_file(socket)
```
Comment (LtChae) [inline apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex:225]: This did not originally set `loading_schema` to false. Not sure if that makes a difference

File: apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex [hunk 5]
```diff
@@ -182,30 +188,28 @@ defmodule AndiWeb.EditLiveView.DataDictionaryForm do
...
-          |> DataDictionaryFormSchema.changeset_from_file(socket.assigns.dataset_id)
-
-        assign_new_schema(socket, new_changeset)
+        try do
+          new_changeset =
+            decoded_json
+            |> Enum.empty?()
+            |> if(do: throw(:error), else: decoded_json)
+            |> List.wrap()
+            |> DataDictionaryFormSchema.changeset_from_file(socket.assigns.dataset_id)
```
Comment (LtChae) [inline apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex:201]: This is an unconventional pattern. Feels weird to swap the original value of the pipeline back into it further down.

File: apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex (hunk 2 above)
Comment (bbrewer-forge) [reply apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex:174]: +1

**Review by LtChae — COMMENTED** [review]
See comments

File: apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex (hunk 1 above)
Comment (sukaant) [reply apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex:173]: For empty CSV files, I was getting "\n".

**Review by JarredOlson — APPROVED** [review]

**Pull request merged**